components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAO.java

```diff
@@ -416,6 +416,45 @@ public void deleteIdP(String idPName, int tenantId, String tenantDomain)
 
     }
 
+
+    public void forceDeleteIdP(String idPName, int tenantId, String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("------ Force deleting IDP:%s of tenantDomain:%s -----", idPName, tenantDomain));
+            log.debug("Removing entry for Identity Provider " + idPName + " from cache");
+        }
+
+        IdentityProvider identityProvider = this.getIdPByName(null, idPName, tenantId,
+                tenantDomain);
+        IdPNameCacheKey idPNameCacheKey = new IdPNameCacheKey(idPName, tenantDomain);
+        idPCacheByName.clearCacheEntry(idPNameCacheKey);
+
+        if (identityProvider.getHomeRealmId() != null) {
+            IdPHomeRealmIdCacheKey idPHomeRealmIdCacheKey = new IdPHomeRealmIdCacheKey(
+                    identityProvider.getHomeRealmId(), tenantDomain);
+            idPCacheByHRI.clearCacheEntry(idPHomeRealmIdCacheKey);
+        }
+
+        if (identityProvider.isPrimary()) {
+            primaryIdPs.remove(tenantDomain);
+        }
+
+        if (IdentityApplicationConstants.RESIDENT_IDP_RESERVED_NAME.equals(
+                identityProvider.getIdentityProviderName())) {
+            residentIdPs.remove(tenantDomain);
```

Comment:
Do we allow to force delete resident IDP as well ?

Reply:
Yeah, we allow deleting the Resident IDP in normal delete, So decide to follow the same here.